advisor-ui/src/pages/VolunteerHours/VolunteerHours.jsx

```diff
@@ -0,0 +1,122 @@
+        { params: { studentId: user.id } }
+      );
+      const fetchedStudentEvents = res.data.studentEvents;
+      setStudentEvents(fetchedStudentEvents);
+    } catch (error) {
+      alert("Something went wrong!");
+    }
+  };
+
+  useEffect(() => {
+    fetchStudentEvents();
+  }, []);
+
+  const deleteStudentEventFromState = (id) => {
+    const filteredEvents = studentEvents.filter(
+      (studentEvent) => studentEvent.id !== id
+    );
+    setStudentEvents(filteredEvents);
+  };
+
+  const deleteStudentEventFromDB = async (id) => {
+    try {
+      const res = axios.delete(
+        `http://localhost:5000/api/student-event/${id}`,
+        { params: { studentEventId: id } }
+      );
+      return true;
+    } catch (error) {
+      alert("Unable to delete. Try again later.");
+      return false;
+    }
+  };
+
+  const handleDeleteStudentEvent = async (id) => {
+    // step 1 delete DB record
+    const successfulDelete = await deleteStudentEventFromDB(id);
+    // step 2 delete from local state copy
+    if (successfulDelete) {
+      deleteStudentEventFromState(id);
+    }
+  };
+
+  const createDateFromTimeStamp = (timestamp) => {
+    const date = new Date(timestamp);
+    const year = date.getFullYear();
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    return `${year}-${month}-${day}`;
+  };
+
+  const studentEventRows = studentEvents.map((studentEvent) => {
+    const eventDetails = studentEvent.EventDetail;
+    const hours = studentEvent.hours;
+    return (
+      <TableRow key={studentEvent.id}>
+        <TableCell>{eventDetails.title}</TableCell>
+        <TableCell align="right">
+          {createDateFromTimeStamp(eventDetails.date)}
+        </TableCell>
+        <TableCell align="right">{`${eventDetails.city}, ${eventDetails.state}`}</TableCell>
+        <TableCell align="right">{eventDetails.admin}</TableCell>
+        <TableCell align="right">{hours}</TableCell>
+        <TableCell align="right">
+          <IconButton onClick={() => handleDeleteStudentEvent(studentEvent.id)}>
+            <DeleteIcon />
+          </IconButton>
+        </TableCell>
+      </TableRow>
+    );
+  });
+
+  return (
+    <div className="volunteer-hours">
+      <div className="content">
+        <Link to="/student/volunteer">
+          <ArrowBackIcon className="back" />
+        </Link>
+        <div className="table">
+          <h3>Your Volunteer History:</h3>
+          <TableContainer>
+            <Table sx={{ minWidth: 650 }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell>Title</TableCell>
+                  <TableCell align="right">Date</TableCell>
+                  <TableCell align="right">Location</TableCell>
+                  <TableCell align="right">Admin</TableCell>
+                  <TableCell align="right">Hours</TableCell>
+                  <TableCell align="right">Actions</TableCell>
```

Comment:
One thing to just think about here is if you could create these column headers programmatically, either from a list of columns here on the frontend or by passing in display names for the fields from the backend. It won't make a big deal here, but could allow you to generalize your table logic and reuse it across the app.